### PR TITLE
[SUGGESTION] Simple validation forms suggestion 1

### PR DESCRIPTION
**Pull request by cintamani**
As per conversation with the devs, this is one suggestion of simple forms for validations only

### CODE REVIEW

File: app/forms/charge_adjust_form.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  attr_accessor :charge_type
+
+  validates :charge_type, presence: true
+
+  def submit(params)
+    # Assign the params for validation
+    self.charge_type = params[:charge_type]
+
+    valid?
```
Comment (cintamani) [inline app/forms/charge_adjust_form.rb]: No need for this?

**Comment by Cruikshanks** [comment]
Having compared this with #685 I've decided this is the solution to opt for.

Solution 3 retains the pattern we have previously implemented but I think its reliance on inheriting from `ResourceFormsController` means it brings in a lot of functionality it doesn't need.

I then did suggest to @cintamani , what if we refactored `ResourceFormsController` to a simpler base that `ResourceFormsController` builds on? She rightly pointed out we already have done that with the concerns. The functionality we need already sits in concerns that `ResourceFormsController ` also pulls in.

I think there could be a case for a `SimpleFormsController` that brought in those shared concerns. It would give a clear name for the type of page a developer is working on and some consistency across them.

But those, like the example this suggestion is built, on are so simple, I think it is more likely to run the risk of premature optimization.

So its a big 👍 and ❤️ for this as our way forward.

**Comment by irisfaraway** [comment]
I'm happy to go with option 1 as well.

I think I agree with what @Cruikshanks said about a `SimpleFormsController` (and a `SimpleForm` as well) - if you open up a file and can immediately tell what type of form or controller it is based on the subclassing, it might make things clearer, particularly as we now have so many forms. But also if it starts to tie us back into knots then it's not worth it.

**Comment by cintamani** [comment]
It might be clearer, but I am unsure about it's maintainability. Let's keep that chat for the next ☕️ and 🍰 meeting :) 

Going to close this PRs now :)

**Pull request closed without merging**